### PR TITLE
[v2.7] k8s November 2023 patches

### PR DESCRIPTION
**Pull request by vardhaman22**
**since november patches are released, skipping the october patches.**
related issues: https://github.com/rancher/rancher/issues/43073, https://github.com/rancher/rancher/issues/43152, https://github.com/rancher/rancher/issues/43205,
https://github.com/rancher/rancher/issues/43638, https://github.com/rancher/rancher/issues/43640, https://github.com/rancher/rancher/issues/43213

- adds k8s v1.25.16, v1.26.11 versions for rke1
- for all the new versions uses etcd v3.5.9 and rke-tools v0.1.96
- adds nginx v1.9.4 template and use it in all newly introduced versions.
- for nginx v1.9.4 update ingress-nginx/kube-webhook-certgen image to v20231011-8b53cabe0 [ref](https://github.com/kubernetes/ingress-nginx/blob/5583f90c7f3525b6267747d6b07541c01218a9ea/deploy/static/provider/cloud/deploy.yaml#L591)
- also includes the latest changes of dev-v2.7-2023-09-patches branch (needed because a few more changes went in for september patches)
- bumped calico to v3.26.3 for new k8s versions
- bumped k8s dns images to 1.22.28 for new k8s versions
- included hyperkube-base fix https://github.com/rancher/rancher/issues/43213

~Also, covers https://github.com/rancher/rancher/issues/43002~ Edit: It doesn't because destination branch is not `dev-v2.7`

### CODE REVIEW

**Review by snasovich — CHANGES_REQUESTED** [review]
@vardhaman22 , please update `rke-tools` to `v0.1.96`. While at it, could you please also update it in previous versions as done per https://github.com/rancher/kontainer-driver-metadata/pull/1259 (in fact, please bring in other changes that got into `dev-v2.7-2023-09-patches` branch after we branched for `dev-v2.7-2023-10-patches` - e.g. RKE2/k3s R2 releases).

**Comment by vardhaman22** [comment]
> @vardhaman22 , please update `rke-tools` to `v0.1.96`. While at it, could you please also update it in previous versions as done per #1259 (in fact, please bring in other changes that got into `dev-v2.7-2023-09-patches` branch after we branched for `dev-v2.7-2023-10-patches` - e.g. RKE2/k3s R2 releases).

rebased my branch with  `dev-v2.7-2023-09-patches`  and also updated rke-tools image to v0.1.96 in the latest changes.

File: pkg/rke/templates/calico_v3.36.go
Comment (kinarashah) [inline pkg/rke/templates/calico_v3.36.go:1]: file name: calico_v3.26.go

Comment (vardhaman22) [reply pkg/rke/templates/calico_v3.36.go:1]: updated

**Review by kinarashah — DISMISSED** [review]
- confirmed the k8s system images are same as v2.8 PR 
- confirmed default k8s version 
- confirmed addon versions in templates.go

**Review by snasovich — DISMISSED** [review]

**Comment by snasovich** [comment]
LGTM. From what I can see this includes ensuring we have changes that were released per Sep patches into this branch too.

Also, I have removed references to 2.8 issues in the description of this PR.

**Review by snasovich — APPROVED** [review]
LGTM, thank you for addressing https://github.com/rancher/kontainer-driver-metadata/pull/1256#discussion_r1416611256

**Review by kinarashah — APPROVED** [review]

**Pull request merged**